src/coreComponents/mesh/CellElementSubRegion.cpp

```diff
@@ -320,6 +336,8 @@ void CellElementSubRegion::
                               m_elementType, getName() ) );
     }
   }
+
+  LvArray::tensorOps::scale< 3 >( m_elementCenter[ k ], 1.0 / m_numNodesPerElement );
```

Comment:
As a general comment, I think the existence of this single-cell function `calculateElementCenterAndVolume` as a (non-static) member of `CellElementSubRegion` is a bit weird, and prevents us from moving this computation on device, if we wanted to. It's only ever called from inside the kernel launch of `calculateElementGeometricQuantities` but has to repeatedly switch on element type (which is constant).

Ideally I would like to:
1. Do the `switch` in `calculateElementGeometricQuantities`, outside of the kernel launch.
2. Inside each `case`, delegate to a helper free function, say `elementGeometricQuantitiesKernel`, templated on `ElemenType`, passing input views `X` and `m_toNodesRelation` along with output views `m_elementVolume` and `m_elementCenter`.
3. `elementGeometricQuantitiesKernel` does the kernel launch (maybe take launch policy as parameter too) and uses compile time knowledge of element shape to allocate `Xlocal` of the right size, unroll loops, etc. as well as delegate to the correct FE type for volume calculation.

(this might require establishing a compile-time mapping between `ElemenType` and the corresponding first-order `finiteElement` class, if one does not exist somewhere in code already).

Perhaps all of this can be addressed separately, to avoid scope creep of this PR.

Reply:
I agree, let's address this in another PR.